nifi-commons/nifi-security-utils/src/test/groovy/org/apache/nifi/security/util/CertificateUtilsTest.groovy

```diff
@@ -272,4 +275,179 @@ class CertificateUtilsTest extends GroovyTestCase {
         assert convertedCertificate instanceof X509Certificate
         assert convertedCertificate == EXPECTED_NEW_CERTIFICATE
     }
+
+    @Test
+    void testShouldDetermineClientAuthStatusFromSocket() {
+        // Arrange
+        SSLSocket needSocket = [getNeedClientAuth: { -> true }] as SSLSocket
```

Comment:
Discussed with Matt offline, but these are standard Groovy map coercions, not lambdas. Running a local check against Java 7 to ensure no Java 8 features were introduced.